platform/measurement_utils.cpp

```diff
@@ -18,12 +18,28 @@
 
 namespace measurement_utils
 {
-std::string ToStringPrecision(double d, int pr)
+// Global Locale variable to store system locale.
+static platform::Locale g_systemLocale = {
```

Comment:
```suggestion
static platform::Locale g_systemLocale =
{
```
C++20 allows to specify which struct members are initialized, for clarity.

Did you also test the desktop version? How will it behave with "undefined" locale values? On desktops (Mac, Linux) locale is properly set when the app is launched.

Reply:
I've checked the Linux desktop version and the distance formatting does not seem to follow system's locale (neither with the previous nor with the new implementation). That problem for desktop shall be managed in a separate issue...